test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt

```diff
@@ -796,4 +1225,270 @@ class QueryTests {
+            }
+            AggregatorQueryType.MAX -> when {
+                returnType.isMarkedNullable -> NULLABLE_DOUBLE_VALUES.mapNotNull { it }.maxOrNull()
+                else -> DOUBLE_VALUES.maxOrNull()
+            }
+            AggregatorQueryType.SUM -> when {
+                returnType.isMarkedNullable -> NULLABLE_DOUBLE_VALUES.sumOf { it ?: 0.0 }
+                else -> DOUBLE_VALUES.sum()
+            }
+        }?.toDouble() ?: throw IllegalArgumentException("Aggregate result cannot be null.")
+        Float::class -> when (type) {
+            AggregatorQueryType.MIN -> when {
+                returnType.isMarkedNullable -> NULLABLE_FLOAT_VALUES.mapNotNull { it }.minOrNull()
+                else -> FLOAT_VALUES.minOrNull()
+            }
+            AggregatorQueryType.MAX -> when {
+                returnType.isMarkedNullable -> NULLABLE_FLOAT_VALUES.mapNotNull { it }.maxOrNull()
+                else -> FLOAT_VALUES.maxOrNull()
+            }
+            AggregatorQueryType.SUM -> when {
+                returnType.isMarkedNullable -> NULLABLE_FLOAT_VALUES.sumOf { it?.toInt() ?: 0 }
+                else -> FLOAT_VALUES.sum()
+            }
+        }?.toFloat() ?: throw IllegalArgumentException("Aggregate result cannot be null.")
+        RealmInstant::class -> when (type) {
+            AggregatorQueryType.MIN -> when {
+                returnType.isMarkedNullable ->
+                    NULLABLE_TIMESTAMP_VALUES.mapNotNull { it }.minOrNull()
+                else -> TIMESTAMP_VALUES.minOrNull()
+            }
+            AggregatorQueryType.MAX -> when {
+                returnType.isMarkedNullable ->
+                    NULLABLE_TIMESTAMP_VALUES.mapNotNull { it }.maxOrNull()
+                else -> TIMESTAMP_VALUES.maxOrNull()
+            }
+            AggregatorQueryType.SUM -> throw IllegalArgumentException("SUM is not allowed on timestamp fields.")
+        } ?: throw IllegalArgumentException("Aggregate result cannot be null.")
+        else -> throw IllegalArgumentException("Only numerical properties and timestamps are allowed.")
+    }
+
+    // ------------------------
+    // General-purpose helpers
+    // ------------------------
+
+    // Deletes all objects after assertions to avoid "null vs 0" results when testing aggregators
+    private fun cleanUpBetweenProperties() = realm.writeBlocking {
+        query(QuerySample::class)
+            .find { results ->
+                results.toList() // We cannot iterate over the results and delete directly!
+                    .forEach { sample ->
+                        val latest = findLatest(sample)
+                        assertNotNull(latest)
+                        delete(latest)
+                    }
+            }
+    }
+
+    // ----------------------------------------
+    // Descriptors used for exhaustive testing
+    // ----------------------------------------
+
+    private val basePropertyDescriptors = listOf(
```

Comment:
Maybe somehow use the `TypeDescriptor.classifier` as root of making the test exhaustive, so that when introducing a new type, we would have a failing tests that required to add the new type to a set of unsupported aggregator types. Don't know if it makes sense, but would be nice if we somehow ensured that we had to evaluate if a new type was applicable for a certain feature. Otherwise we forget to add it somewhere.

Reply:
I have split `TypeDescriptor.classifiers` into two categories now:
```
    // Classifiers for types that can be used in aggregated queries
    val aggregateClassifiers: Map<KClassifier, CoreFieldType> = mapOf(
        Byte::class to CoreFieldType.INT,
        Char::class to CoreFieldType.INT,
        Short::class to CoreFieldType.INT,
        Int::class to CoreFieldType.INT,
        Long::class to CoreFieldType.INT,
        Float::class to CoreFieldType.FLOAT,
        Double::class to CoreFieldType.DOUBLE
    )

    // Kotlin classifier to Core field type mappings
    val classifiers: Map<KClassifier, CoreFieldType> = aggregateClassifiers + mapOf(
        Boolean::class to CoreFieldType.BOOL,
        String::class to CoreFieldType.STRING,
        RealmInstant::class to CoreFieldType.TIMESTAMP,
        RealmObject::class to CoreFieldType.OBJECT
    )
```
and now I'm using the `aggregateClassifiers` in here instead, mapping them to what I need.

This makes it more failproof. However, I would like to give it another spin at some other point. I think we could make it even more granular.

Reply:
I guess that works. At least they are somehow linked now. 

Just input for future experimens/modeling: Maybe it should even be a boolean attribute on the CoreFieldType so that we could just iterate the classifiers and test those where the corresponding core-type supports aggregation.

Reply:
That's a very good point, but implementing a one-size-fits-all solution is tricky because not all aggregators work for all types, e.g. it's possible to compute the `max`/`min` of a `Timestamp` field but not its `sum` 😭 - that's why I used `propertyDescriptorsForSum` and `propertyDescriptors` in the tests.